src/jit/compiler.cpp

```diff
@@ -4999,8 +4999,12 @@ int           Compiler::compCompileHelper (CORINFO_MODULE_HANDLE            clas
             (fgBBcount > 5) &&
             !forceInline)
         {
-            compInlineResult->setNever("Too many basic blocks in the inlinee");
-            goto _Next;
+            compInlineResult->note(InlineObservation::CALLEE_TOO_MANY_BASIC_BLOCKS);
+
+            if (compInlineResult->isFailure()) 
+            {
+                goto _Next;
+            }
```

Comment:
This is something of an interim state. There are just the two patterns:
```
noteFatal(...);
return;
```
and 
```
note(...)
if (isFailure()) return;
```
Over time more (perhaps all) observation sites will change over to the second pattern, as we feel comfortable allowing policies to defer failing

But for now I wanted to make it simpler to ensure that fatal observations could not be deferred.

FWIW I just tracked down my desktop testing bug.... there's one place in the changes above where I failed to follow the first pattern properly, and hit an assert as a result.